common/worker_versioning/worker_versioning.go

```diff
@@ -41,14 +42,43 @@ import (
 )
 
 const (
-	buildIdSearchAttributePrefixAssigned    = "assigned"
-	buildIdSearchAttributePrefixVersioned   = "versioned"
-	buildIdSearchAttributePrefixUnversioned = "unversioned"
-	BuildIdSearchAttributeDelimiter         = ":"
+	buildIdSearchAttributePrefixReachability = "reachability"
+	buildIdSearchAttributePrefixAssigned     = "assigned"
+	buildIdSearchAttributePrefixVersioned    = "versioned"
+	buildIdSearchAttributePrefixUnversioned  = "unversioned"
+	BuildIdSearchAttributeDelimiter          = ":"
 	// UnversionedSearchAttribute is the sentinel value used to mark all unversioned workflows
 	UnversionedSearchAttribute = buildIdSearchAttributePrefixUnversioned
 )
 
+// escapeBuildIdSearchAttributeDelimiter is a helper which escapes the BuildIdSearchAttributeDelimiter character in the input string
+func escapeBuildIdSearchAttributeDelimiter(s string) string {
+	s = strings.Replace(s, BuildIdSearchAttributeDelimiter, `|`+BuildIdSearchAttributeDelimiter, -1)
+	return s
+}
+
+// ReachabilityBuildIdSearchAttribute returns the search attribute value for the currently assigned build ID in the form
+// 'reachability:<behavior>:<deployment_series_name>:<deployment_build_id>'
+func ReachabilityBuildIdSearchAttribute(behavior enumspb.VersioningBehavior, deployment *deploymentpb.Deployment) string {
+	var escapedDeployment string
+	if deployment == nil {
+		escapedDeployment = "UNVERSIONED"
+	} else {
+		escapedDeployment = fmt.Sprintf("%s%s%s",
+			escapeBuildIdSearchAttributeDelimiter(deployment.GetSeriesName()),
+			BuildIdSearchAttributeDelimiter,
+			escapeBuildIdSearchAttributeDelimiter(deployment.GetBuildId()),
+		)
+	}
+	return sqlparser.String(sqlparser.NewStrVal([]byte(fmt.Sprintf("%s%s%s%s%s",
+		buildIdSearchAttributePrefixReachability,
+		BuildIdSearchAttributeDelimiter,
+		escapeBuildIdSearchAttributeDelimiter(behavior.String()),
```

Comment:
would this be a short "PINNED" string on full enum name "VERSIONING_BEHAVIOR_PINNED"? short is better.

Reply:
the strings are "Pinned", "AutoUpgrade", and "Unspecified", so they're reasonably short